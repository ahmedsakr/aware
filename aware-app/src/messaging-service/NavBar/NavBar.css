#navigation-menu {
  background-color: #232f3e;
  position: relative;
}

.navigation-menu-title {
  color: orange;
}

.navigation-menu-logo > p {
  color: white;
  display: inline;
  font-size: 1.5em;
}

.navigation-menu-link > p {
  color: white;
  display: inline;
  padding-left: 5px;
  font-size: 0.8em;
}

.navigation-menu-link > span {
  display: inline;
  color: white;
}

.navigation-menu-link-active {
  background-color: #18202c;
}

@media screen and (max-height: 480px) {
  .navigation-menu-logo, .navigation-menu-link, .navigation-menu-title {
    display: inline-block;
    height: 50px;
    line-height: 50px;
    vertical-align: middle;
  }
}

@media screen and (max-height: 800px) {
  .navigation-menu-logo, .navigation-menu-link, .navigation-menu-title {
    display: inline-block;
    height: 70px;
    line-height: 70px;
    vertical-align: middle;
  }
}