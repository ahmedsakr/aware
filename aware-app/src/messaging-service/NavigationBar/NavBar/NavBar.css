#navigation-menu {
  background-color: #232f3e;
  display: block;
  text-align: center;
}

.navigation-menu-link-active {
  background-color: #18202c;
}

.navigation-menu-group {
  display: inline-block;
}

.overlay-div {
  display: inline-block;
}