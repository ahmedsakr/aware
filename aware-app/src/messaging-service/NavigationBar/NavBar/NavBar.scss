/*
 * NavBar styling
 *
 * Defines layout fundamentals for the NavBar.
 */

#navigation-menu {
    background-color: #232f3e;
    display: block;
    text-align: center;
}

.navigation-menu-link-active {
    background-color: #18202c;
}

.navigation-menu-group {
    display: inline-block;
}

.overlay-div {
    display: inline-block;
}