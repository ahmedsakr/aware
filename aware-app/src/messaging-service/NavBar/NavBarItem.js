import React, {Component} from 'react';
import './NavBar.css'
class NavBarLink extends Component {
  render() {
    const {className, icon, name} = this.props;

    return (
      <a href="/" class={className + " navigation-menu-link"}>
        <span class={icon} aria-hidden="true"></span>
        <p>{name}</p>
      </a>
    );
  }
}

export default NavBarLink;

class NavBarLogo extends Component {
  render() {
    const {className} = this.props;

    return (
      <div class={className + " navigation-menu-logo"}>
        <p>AWARE</p>
      </div>
    );
  }
}

class NavBarTitle extends Component {
  render() {

    const {className, title} = this.props;

    return (
      <div class={className + " navigation-menu-title"}>
        <p>{title}</p>
      </div>
    );
  }
}

export {
  NavBarLogo,
  NavBarTitle
}