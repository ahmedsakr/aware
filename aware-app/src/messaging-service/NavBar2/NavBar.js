import React, {Component} from 'react';
import NavBarLink, {NavBarLogo, NavBarTitle} from './NavBarItem'
import './NavBar.css';

class NavBar extends Component {
  render() {
    return (
      <div id="navigation-menu">
        <NavBarLogo/>
        <NavBarLink active="false" icon="fa fa-home" name="Home" />
        <NavBarLink active="true" icon="fa fa-comments" name="Messenger" />
        <NavBarLink active="false" icon="fa fa-bars" name="Forum" />
        <NavBarTitle title="SYSC 2100 - Course Discussion"/>
        <NavBarLink className="offset-2" active="false" icon="fa fa-cogs" name="Settings" />
        <NavBarLink active="false" icon="fa fa-user-circle" name="Account" />
      </div>
    );
  }
}

export default NavBar;
