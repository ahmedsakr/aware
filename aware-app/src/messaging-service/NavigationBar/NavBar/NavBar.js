import React, {Component} from 'react';
import NavBarLink, {NavBarLogo, NavBarTitle} from '../NavBarItem/NavBarItem'
import './NavBar.css';

class NavBar extends Component {
  render() {
    return (
      <div id="navigation-menu">
        <NavBarLogo className="col-lg-2 col-md-2 col-sm-2 p-0"/>
        <div class="col-lg-3 col-md-4 col-sm-5 p-0 navigation-menu-group">
          <NavBarLink className="col-lg-3 col-md-4 col-sm-4 p-0" icon="fa fa-home" name="Home" />
          <NavBarLink className="col-lg-5 col-md-4 col-sm-4 navigation-menu-link-active p-0" icon="fa fa-comments" name="Messenger" />
          <NavBarLink className="col-lg-4 col-md-4 col-sm-4 p-0" icon="fa fa-bars" name="Forum" />
        </div>

        <NavBarTitle className="col-lg-5 col-md-4 col-sm-2 p-0" title={this.props.activeRoom}/>
        <div class="col-lg-2 col-md-2 col-sm-3 p-0 navigation-menu-group">
          <NavBarLink className="col-lg-6 col-md-6 col-sm-6 p-0" icon="fa fa-cogs" name="Settings" />
          <NavBarLink className="col-lg-6 col-md-6 col-sm-6 p-0" icon="fa fa-user-circle" name="Account" />
        </div>
      </div>
    );
  }
}

export default NavBar;
