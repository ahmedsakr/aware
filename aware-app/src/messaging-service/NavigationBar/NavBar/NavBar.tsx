import React from 'react';
import NavBarLink, { NavBarLogo, NavBarTitle } from '../NavBarLink/NavBarLink'
import './NavBar.css';

type NavBarProps = {
    activeRoom: string
};

type NavBarState = {};

export default class NavBar extends React.Component<NavBarProps, NavBarState> {
    render(): JSX.Element {
        return (
            <div id="navigation-menu">
                <NavBarLogo className="col-lg-2 col-md-2 col-sm-2 p-0" />
                <div className="col-lg-3 col-md-4 col-sm-5 p-0 navigation-menu-group">
                    <NavBarLink className="col-lg-3 col-md-4 col-sm-4 p-0" icon="fa fa-home" name="Home" />
                    <NavBarLink className="col-lg-5 col-md-4 col-sm-4 navigation-menu-link-active p-0" icon="fa fa-comments" name="Messenger" />
                    <NavBarLink className="col-lg-4 col-md-4 col-sm-4 p-0" icon="fa fa-bars" name="Forum" />
                </div>

                <NavBarTitle className="col-lg-5 col-md-4 col-sm-2 p-0" title={this.props.activeRoom} />
                <div className="col-lg-2 col-md-2 col-sm-3 p-0 navigation-menu-group">
                    <div className="col-lg-6 col-md-6 col-sm-6 p-0 overlay-div" data-toggle="modal" data-target={"#newsletter-overlay"}>
                        <NavBarLink className="col-12 p-0" icon="fa fa-cogs" name="Settings" />
                    </div>

                    <NavBarLink className="col-lg-6 col-md-6 col-sm-6 p-0" icon="fa fa-user-circle" name="Account" />
                </div>
            </div>
        );
    }
}
