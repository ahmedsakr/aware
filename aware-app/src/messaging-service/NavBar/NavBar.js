import React, {Component} from 'react';
import NavBarItem from './NavBarItem'
import './NavBar.css';

class NavBar extends Component {
  render() {
    return (
      <div id="vertical-menu" class="row">
        <NavBarItem src="/icons8-monitor-64.png" name="sysc 2100" />
        <NavBarItem src="/icons8-monitor-64.png" name="sysc 2003" />
        <NavBarItem src="/icons8-monitor-64.png" name="sysc 2004" />
        <NavBarItem src="/icons8-electronics-50.png" name="elec 2501" />
        <NavBarItem src="/icons8-electronics-50.png" name="elec 2607" />
      </div>
    );
  }
}

export default NavBar;
