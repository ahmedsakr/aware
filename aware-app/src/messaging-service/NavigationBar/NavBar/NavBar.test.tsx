import React from 'react';
import { shallow } from 'enzyme';

import NavBar from './NavBar';

describe('NavBar', () => {
    it('should render correctly', () => {

        const component = shallow(
            <NavBar
                activeRoom="SYSC 2100" />
        );

        expect(component).toMatchSnapshot();
    });
});