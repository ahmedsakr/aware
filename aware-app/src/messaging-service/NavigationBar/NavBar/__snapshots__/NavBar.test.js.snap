// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly in "debug" mode 1`] = `
<div
  id="navigation-menu"
>
  <NavBarLogo
    className="col-lg-2 col-md-2 col-sm-2 p-0"
  />
  <div
    class="col-lg-3 col-md-4 col-sm-5 p-0 navigation-menu-group"
  >
    <NavBarLink
      className="col-lg-3 col-md-4 col-sm-4 p-0"
      icon="fa fa-home"
      name="Home"
    />
    <NavBarLink
      className="col-lg-5 col-md-4 col-sm-4 navigation-menu-link-active p-0"
      icon="fa fa-comments"
      name="Messenger"
    />
    <NavBarLink
      className="col-lg-4 col-md-4 col-sm-4 p-0"
      icon="fa fa-bars"
      name="Forum"
    />
  </div>
  <NavBarTitle
    className="col-lg-5 col-md-4 col-sm-2 p-0"
  />
  <div
    class="col-lg-2 col-md-2 col-sm-3 p-0 navigation-menu-group"
  >
    <div
      className="col-lg-6 col-md-6 col-sm-6 p-0 overlay-div"
      data-target="#newsletter-overlay"
      data-toggle="modal"
    >
      <NavBarLink
        className="col-12 p-0"
        icon="fa fa-cogs"
        name="Settings"
      />
    </div>
    <NavBarLink
      className="col-lg-6 col-md-6 col-sm-6 p-0"
      icon="fa fa-user-circle"
      name="Account"
    />
  </div>
</div>
`;
